pandas/core/indexes/datetimes.py

```diff
@@ -2269,18 +2269,27 @@ def _generate_regular_range(start, end, periods, offset):
         data = DatetimeIndex._simple_new(data, None, tz=tz)
     else:
         if isinstance(start, Timestamp):
-            start = start.to_pydatetime()
```

Comment:
hmm why are you doing this here? if its necessary to convert to a datetime (why is that?) should do it generate_range